airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java

```diff
@@ -712,7 +709,141 @@ public void testResetCancelsRunningSync() throws Exception {
+
+    testHarness.assertSourceAndDestinationDbInSync(false);
+    assertEquals(Jsons.deserialize(expectedState), state.getState());
+  }
+
+  @Test
+  public void testSyncAfterUpgradeToPerStreamStateWithNoNewData(final TestInfo testInfo) throws Exception {
+    LOGGER.info("Starting {}", testInfo.getDisplayName());
+    final String connectionName = "test-connection";
+    final SourceRead source = testHarness.createPostgresSource();
+    final UUID sourceId = source.getSourceId();
+    final UUID sourceDefinitionId = source.getSourceDefinitionId();
+    final UUID destinationId = testHarness.createDestination().getDestinationId();
+    final UUID operationId = testHarness.createOperation().getOperationId();
+    final AirbyteCatalog catalog = testHarness.discoverSourceSchema(sourceId);
+
+    // Set the source to a version that does not support per-stream state
+    LOGGER.info("Setting source connector to pre-per-stream state version {}...",
+        AirbyteAcceptanceTestHarness.POSTGRES_SOURCE_LEGACY_CONNECTOR_VERSION);
+    testHarness.updateSourceDefinitionVersion(sourceDefinitionId, AirbyteAcceptanceTestHarness.POSTGRES_SOURCE_LEGACY_CONNECTOR_VERSION);
+
+    catalog.getStreams().forEach(s -> s.getConfig()
+        .syncMode(INCREMENTAL_SYNC_MODE)
+        .cursorField(List.of(COLUMN_ID))
+        .destinationSyncMode(DESTINATION_SYNC_MODE_APPEND));
+    final UUID connectionId =
+        testHarness.createConnection(connectionName, sourceId, destinationId, List.of(operationId), catalog, null).getConnectionId();
+    LOGGER.info("Beginning {} sync 1", testInfo.getDisplayName());
+
+    final JobInfoRead connectionSyncRead1 = apiClient.getConnectionApi()
+        .syncConnection(new ConnectionIdRequestBody().connectionId(connectionId));
+    waitForSuccessfulJob(apiClient.getJobsApi(), connectionSyncRead1.getJob());
+    LOGGER.info("state after sync 1: {}", apiClient.getConnectionApi().getState(new ConnectionIdRequestBody().connectionId(connectionId)));
+
+    testHarness.assertSourceAndDestinationDbInSync(false);
+
+    // Set source to a version that supports per-stream state
+    testHarness.updateSourceDefinitionVersion(sourceDefinitionId, AirbyteAcceptanceTestHarness.POSTGRES_SOURCE_PER_STREAM_STATE_CONNECTOR_VERSION);
+    LOGGER.info("Upgraded source connector per-stream state supported version {}.",
+        AirbyteAcceptanceTestHarness.POSTGRES_SOURCE_PER_STREAM_STATE_CONNECTOR_VERSION);
+
+    // sync one more time. verify that nothing has been synced
+    LOGGER.info("Starting {} sync 3", testInfo.getDisplayName());
+    final JobInfoRead connectionSyncRead3 =
+        apiClient.getConnectionApi().syncConnection(new ConnectionIdRequestBody().connectionId(connectionId));
+    waitForSuccessfulJob(apiClient.getJobsApi(), connectionSyncRead3.getJob());
+    LOGGER.info("state after sync 3: {}", apiClient.getConnectionApi().getState(new ConnectionIdRequestBody().connectionId(connectionId)));
```

Comment:
Bad copy paste - should be `sync 2` and `connectionSyncRead2` instead of `sync 3`/`connectionSyncRead3`

Reply:
I think what happened is that sync 2 got removed from this test.  Will rename so it is more consistent/clear.